### PR TITLE
Fixing product-iots#1313

### PR DESCRIPTION
**Pull request by dilee**
Fixed https://github.com/wso2/product-iots/issues/1313

### CODE REVIEW

**Review by ruwany — APPROVED** [review]

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js [hunk 1]
```diff
@@ -431,3 +431,76 @@ function viewFence(geoFenceElement,id) {
...
+            popupTemplate.find('#exportGeoJson').attr('leaflet_id', geometryShape._leaflet_id);
+            popupTemplate.find('#hideViewFence').attr('leaflet_id', geometryShape._leaflet_id);
+            popupTemplate.find('#viewAreaName').html(areaName);
+            popupTemplate.find('#exitAlertForm').attr('area-name', areaName);
+            popupTemplate.find('#exitAlertForm').attr('query-name', queryName);
+            geometryShape.bindPopup(popupTemplate.html(), {closeButton: true}).openPopup();
+            // transparent the layer .leaflet-popup-content-wrapper
+            $(geometryShape._popup._container.childNodes[0]).css("background", "rgba(255,255,255,0.8)");
+        });
+    } else if(id=="Traffic"){
```
Comment (madawas) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js:502]: @dilee is there any reason to have this empty if block? Also, I found some formatting issues in line 476, 489, 502. Can you check and fix these?

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js [hunk 2]
```diff
@@ -431,3 +431,76 @@ function viewFence(geoFenceElement,id) {
...
+
+function viewFenceByData(geoJson, queryName, areaName, stationeryTime, id) {
+    var matchResults = /(?:"geoFenceGeoJSON"):"{(.*)}"/g.exec(geoJson);
+    if (matchResults && matchResults.length > 1) {
+        geoJson = "{" + matchResults[1] + "}";
+    }
+    geoJson = JSON.parse(geoJson.replace(/'/g, '"'));
+    var geometryShape;
+
+    if(geoJson.type=="Point"){
```
Comment (Kamidu) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js:443]: Apply the code formatting to the code.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js (hunk 1 above)
Comment (dilee) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js:502]: Fixed in https://github.com/wso2/carbon-device-mgt/pull/950

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js (hunk 2 above)
Comment (dilee) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js:443]: Fixed in https://github.com/wso2/carbon-device-mgt/pull/950

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js [hunk 3]
```diff
@@ -431,3 +431,76 @@ function viewFence(geoFenceElement,id) {
...
+
+        var circleOptions = {
+            color: '#ff0043'
+        };
+        geometryShape= new L.circle([geoJson.coordinates[1],geoJson.coordinates[0]], geoJson.radius,circleOptions);
+        // var marker=new L.marker([geoJson.coordinates[1],geoJson.coordinates[0]]);
+        map.addLayer(geometryShape);
+        //  map.addLayer(marker);
```
Comment (prabushi) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js:451]: Remove commented lines. (#449 and #451)

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js [hunk 4]
```diff
@@ -431,3 +431,76 @@ function viewFence(geoFenceElement,id) {
...
+    } else if(geoJson.type=="Polygon"){
```
Comment (prabushi) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js]: Fix formatting issue.

Comment (prabushi) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js]: Fixed the formatting issues in https://github.com/wso2/carbon-device-mgt/pull/950

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js (hunk 3 above)
Comment (dilee) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.geo-dashboard/public/js/geo_fencing.js:451]: Fixed in https://github.com/wso2/carbon-device-mgt/pull/1275

**Pull request merged**